docs/guides/using-rewrites.md

```diff
@@ -0,0 +1,188 @@
+# Rewriting paths
+This guide demonstrates host and path rewrites using Ingress and Service configuration.
+
+## Installation
+
+Please follow the [deployment](../deployment) documentation to install
+Kong Ingress Controller on your Kubernetes cluster.
+
+## Testing Connectivity to Kong
+
+This guide assumes that the `PROXY_IP` environment variable is
+set to contain the IP address or URL pointing to Kong.
+Please follow one of the
+[deployment guides](../deployment) to configure this environment variable.
+
+If everything is setup correctly, making a request to Kong should return
+HTTP 404 Not Found.
+
+```bash
+$ curl -i $PROXY_IP
+HTTP/1.1 404 Not Found
+Date: Fri, 21 Jun 2019 17:01:07 GMT
+Content-Type: application/json; charset=utf-8
+Connection: keep-alive
+Content-Length: 48
+Server: kong/1.2.1
+
+{"message":"no Route matched with those values"}
+```
+
+This is expected as Kong does not yet know how to proxy the request.
+
+## Create a test Deployment
+
+To test our requests, we create an echo server Deployment, which responds to
+HTTP requests with a summary of the request contents:
+
+```bash
+$ kubectl create namespace echo
+namespace/echo created
+$ kubectl apply -n echo -f https://bit.ly/echo-service
+service/echo created
+deployment.apps/echo created
+```
+
+## Create a Kubernetes service
+
+First, create a Kubernetes Service:
+
+```bash
+echo "
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+  namespace: echo
+spec:
+  selector:
+    app: echo
+  ports:
+    - name: http
+      protocol: TCP
+      port: 80
+      targetPort: 80
+" | kubectl create -f -
+```
+
+This Service will create a Kong service and upstream that uses the upstream IPs
+(Pod IPs) for its `Host` header and appends request paths starting at `/`.
```

Comment:
This statement threw me off a little. The service in Kong is not created until Ingress resource is created.
Should we move this statement under the next statement?
We can also explain here that this is the default behavior if you don't add in any annotation whatsoever.